chain/chain/src/types.rs

```diff
@@ -171,7 +171,9 @@ impl BlockEconomicsConfig {
     const MAX_GAS_MULTIPLIER: u128 = 20;
     /// Compute min gas price according to protocol version and genesis protocol version.
     pub fn min_gas_price(&self, protocol_version: ProtocolVersion) -> Balance {
-        if self.genesis_protocol_version < MIN_PROTOCOL_VERSION_NEP_92 {
+        if cfg!(feature = "protocol_feature_lower_min_gas_price") {
```

Comment:
This should use the checked_feature macro:

https://github.com/near/nearcore/blob/83a08f5b02f090363dd21f8601f6b0165f651875/core/primitives/src/version.rs#L251

Even when compile-time feature is enabled, if the runtime protocol_version is low, we should use the old logic.

Reply:
+1

The general behaviour should be:

if your genesis is 'newer' then the protocol version where you introduce this feature -- just use the min_gas_price (as we don't want to overwrite the gas price for 'future' chains)

and let's update/create the tests for these cases.